proxy/src/lib.rs

```diff
@@ -229,20 +231,16 @@ where
 
         let bind = Bind::new().with_sensors(sensors.clone());
 
-        // TODO: Load the TLS configuration asynchronously and watch for
-        // changes to the files.
-        let tls_config = config.tls_settings.and_then(|settings| {
-            match tls::CommonConfig::load_from_disk(&settings) {
-                Ok(config) => Some(config),
-                Err(e) => {
-                    // Keep going without TLS if loading settings failed.
-                    error!("Error loading TLS configuration: {:?}", e);
-                    None
-                }
-            }
-        });
+        let tls_config_changes = config.tls_settings
+            // TODO: determine the correct interval for this.
+            .map(|settings| settings.stream_changes(Duration::from_secs(1)));
```

Comment:
It's going to suck to have this running on our laptops draining battery. We should try to get the inotify-based approach working soon, so that we can avoid this.

Reply:
Yeah, I spent some time this weekend looking into this, and it seems like the [`inotify-rs`](https://crates.io/crates/inotify) library has matured a bit since the last time we looked into it. In particular, it now supports getting watch events as a `futures::Stream`. Therefore, adding `inotify` support in a subsequent PR should be pretty easy now!